lib/utils/location.js

```diff
@@ -39,6 +39,8 @@ export const regexes: { [string]: RegExp } = {
 	domain:           /^\/domain\/([\w\.]+)(?:\/|$)/i,
 	composeMessage:   /^\/(?:r\/([\w\.\+]+)\/)?message\/compose(?:\/|$)/i,
 	liveThread:       /^\/live\/(?!create(?:\/|$))([a-z0-9]+)(?:\/|$)/i,
+	newModMail:       /^\/mail\/(?:all|new|inprogress|archived|highlighted|mod|notifications)(?:\/([0-9a-z]+))?\/?$/i,
```

Comment:
I haven't yet explicitly double-checked this but it looks pretty comprehensive and safe.